plugins/spoiler-alert/assets/javascripts/initializers/spoiler-alert.js

```diff
@@ -0,0 +1,68 @@
+import ComposerController from "discourse/controllers/composer";
+import { withPluginApi } from "discourse/lib/plugin-api";
+import {
+  addBlockDecorateCallback,
+  addTagDecorateCallback,
+} from "discourse/lib/to-markdown";
+import applySpoiler from "discourse/plugins/discourse-spoiler-alert/lib/apply-spoiler";
+
+function spoil(element) {
+  element.querySelectorAll(".spoiler").forEach((spoiler) => {
+    spoiler.classList.remove("spoiler");
+    spoiler.classList.add("spoiled");
+    applySpoiler(spoiler);
+  });
+}
+
+export function initializeSpoiler(api) {
+  api.decorateCookedElement(spoil, { id: "spoiler-alert" });
+
+  api.addToolbarPopupMenuOptionsCallback(() => {
```

Comment:
This has been renamed: https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/lib/plugin-api.js#L761 

Ok to followup with a refactor in a separate PR though!